archunit/src/main/java/com/tngtech/archunit/core/domain/JavaClass.java

```diff
@@ -1063,6 +1083,14 @@ public JavaConstructor getConstructor(String... parameters) {
         return javaClassDependencies.getAnnotationsWithTypeOfClass();
     }
 
+    /**
+     * @return Whether this class has actually been imported (or is just a stub).
+     */
+    @PublicAPI(usage = ACCESS)
+    public boolean isImported() {
```

Comment:
This will be a nice addition I think :smiley: And it will finally make it possible to identify non-stubs in a safe way.